packages/mobile/src/web3/saga.ts

```diff
@@ -230,8 +235,34 @@ export async function readPrivateKeyFromLocalDisk(
 ): Promise<string> {
   const filePath = getPrivateKeyFilePath(account)
   Logger.debug('readPrivateKeyFromLocalDisk', `Reading private key from ${filePath}`)
-  // TODO(ashishb): Read and decrypt private key instead
-  return RNFS.readFile(getPrivateKeyFilePath(account))
+  const hexEncodedEncryptedData: string = await RNFS.readFile(filePath)
+  const encryptedDataBuffer: Buffer = new Buffer(hexEncodedEncryptedData, 'hex')
+  const privateKey: string = getDecryptedData(encryptedDataBuffer, encryptionPassword)
+  ensureAddressAndKeyMatch(account, privateKey)
+  return privateKey
+}
+
+// Exported for testing
+export function getEncryptedData(plainTextData: string, password: string): Buffer {
+  try {
+    const cipher = Crypto.createCipher('aes-256-cbc', password)
+    return Buffer.concat([cipher.update(new Buffer(plainTextData, 'utf8')), cipher.final()])
+  } catch (e) {
+    Logger.error('getEncryptedData', 'Failed to write private key')
```

Comment:
use `TAG + ` in these error messages and include e as the 3rd param

Reply:
Done.